search-app/src/cmr/search/services/humanizers/humanizer_report_service.clj

```diff
@@ -95,3 +105,88 @@
                        "concat humanized rows:" t3))))]
       (info "Create report " t4)
       (str string-writer))))
+
+(defn safe-generate-humanizers-report-csv
+  "This convenience function wraps the report generator in a try/catch."
+  [context]
+  (try
+    (generate-humanizers-report-csv context)
+    (catch Exception e
+      (warn (.getMessage e) "Returning empty report."))))
+
+(def report-cache-key
+  "The key used to store the humanizer report cache in the system cache map."
+  :report-cache)
+
+(defn get-cache
+  "A utility function for returning the system report cache instance."
+  [context]
+  (get-in context [:system :caches report-cache-key]))
+
+(defn read-report
+  "A convenience function wrapping the cache object to retreive report
+  data."
+  [context]
+  (.get-value (get-cache context)
+              csv-report-cache-key))
+
+(defn write-report
+  "A convenience function wrapping the cache object to retreive report
+  data."
+  [context]
+  (.set-value (get-cache context)
```

Comment:
Already done :-) I had the same thought -- I just discovered that function yesterday and remembered it again now.